MonoGame.Framework.Content.Pipeline/Serialization/Intermediate/IntermediateWriter.cs

```diff
@@ -24,37 +34,180 @@ internal IntermediateWriter(IntermediateSerializer serializer, XmlWriter xmlWrit
 
         public void WriteExternalReference<T>(ExternalReference<T> value)
         {
-            throw new NotImplementedException();
+            var externalReference = new ExternalReference
+            {
+                ID = "#External" + (_externalReferences.Count + 1),
+                TargetType = typeof(T).FullName,
+                FileName = MakeRelativePath(value.Filename)
+            };
+            _externalReferences.Add(externalReference);
```

Comment:
I suspect that the same external reference can be written multiple times in the same file.  I think you need to look for duplicates and avoid adding them to the list twice.  The tests must be passing because it doesn't test writing the same reference more than once.